### DISCUSSION TITLE
ImageMagick acting weird when pasting images

### DISCUSSION THREAD

**Discussion by FrostShock** [ImageMagick/ImageMagick#5833] [General]
Since using the ImageMagick version 7.1.0 on Windows 10 Home, something strange happens when pasting images.
Sometimes the script pastes images horizontally and sometimes the same script pastes the images vertically.
Sometimes it pastes the images in the specified order, sometimes the same script pastes the images in reverse order.
And sometimes it simply doesn't work - I can't reproduce the error right now but it says something like it can not find the files.

This is my script, it worked very well with version 5 and 6:

    "C:\Program Files\ImageMagick-7.1.0-Q16-HDRI\magick.exe"  "Image1.jpg"  "Image2.jpg"  "Image3.jpg"  "Image4.jpg"  "Image5.jpg" +append 00-joined-h.jpg

Sometimes I have to change "+append" into "-append" and sometimes I have to write the name of the images in the reverse order to obtain the same result.

**Comment by fmw42** [comment]
Please explain in what way do you mean paste?  In what application?  In Imagemagick as in composite?  What is your exact IM version and platform/OS?  Can you provide your command line?

**Reply by FrostShock** [reply]
> Please explain in what way do you mean paste? In what application? In Imagemagick as in composite? What is your exact IM version and platform/OS? Can you provide your command line?

Sorry, I forgot to mention those things, I provided the details now in the question.

**Comment by fmw42** [comment]
> Sometimes I have to change "+append" into "-append" and sometimes I have to write the name of the images in the reverse order to obtain the same result.

I doubt this is Imagemagick.  + appends horizontally, -appends vertically.  Your images will be loaded in the order you put them and Imagemagick will not reverse them unless you ask it to do so.

What is your exact version of Imagemagick 7.1.0.x?

JPG images have a flag in the header that say how the photo should be rotated or flipped. So perhaps that is the issue. If you want the proper corrected orientation, then add -auto-orient to your command after reading the input images.

See https://imagemagick.org/script/command-line-options.php#auto-orient

**Reply by FrostShock** [reply]
Yes I am using version 7.1.0 - The same script worked very well with versions 5 and 6 but not with 7.1.0
I just ran the script and it joined the images vertically.
Then I added the "-auto-orient" and it correctly joined the images horizontally.
I will come back if I will have any issues in the future

Thank you!
